signac/contrib/project.py

```diff
@@ -142,7 +142,12 @@ def __init__(self, config=None):
         self._config = _ProjectConfig(config)
 
         # Ensure that the project id is configured.
-        self.get_id()
+        id_ = self.get_id()
```

Comment:
Use the property instead of the method. And remove this line, just use “if self.id is None” below since the variable isn’t re-used.